src/compiler/scanner.ts

```diff
@@ -1469,6 +1471,7 @@ namespace ts {
                         pos++;
                         return token = SyntaxKind.MinusToken;
                     case CharacterCodes.dot:
+                        tokenFlags |= TokenFlags.PrecedingDot;

```

Comment:
this might need to move right before the `return` if we understand the flag to denote a single dot.